CMakeLists.txt

```diff
@@ -16,7 +16,9 @@ option(BUILD_P11 "Support for key storage in a HSM via PKCS#11" OFF)
 option(BUILD_SOTA_TOOLS "Set to ON to build SOTA tools" OFF)
 option(BUILD_ISOTP "Set to ON to build ISO-TP" OFF)
 option(BUILD_OPCUA "Set to ON to compile with OPC-UA protocol support" OFF)
-option(BUILD_SYSTEMD "Set to ON to compile with systemd additional support" ON)
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    option(BUILD_SYSTEMD "Set to ON to compile with systemd additional support" ON)
+endif()
```

Comment:
Can we do this via autodetection?  Then it will work on Linux systems without libsystemd-dev installed?

Reply:
Yeah that would be nice. I couldn't quite figure out how to do that, but I will give it another go. `find_library` maybe?